extensions/amp-list/0.1/test-e2e/test-load-more-auto.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+describes.endtoend('AMP list load-more=auto', {
+}, async env => {
+  const pageWidth = 800;
+  const pageHeight = 420;
+  let controller;
+  let ampDriver;
+
+  beforeEach(async() => {
+    controller = env.controller;
+    ampDriver = env.ampDriver;
+
+    await controller.navigateTo('http://localhost:8000/test/manual/amp-list/load-more-auto.amp.html');
+    await ampDriver.toggleExperiment('amp-list-load-more', true);
+
+    await controller.setWindowRect({
+      width: pageWidth,
+      height: pageHeight,
+    });
+    await controller.navigateTo(
+        'http://localhost:8000/test/manual/amp-list/load-more-auto.amp.html');
+  });
+
+  it('should render correctly', async() => {
+    const listItems = await controller.findElements('.item');
+    expect(listItems.length).to.equal(2);
+
+    const loader = await controller.findElement('[load-more-loading]');
+    expect(loader).to.not.be.null;
+
+    const loaderDisplay = await controller.getElementCssValue(loader,
+        'display');
+    expect(loaderDisplay).to.equal('none');
+
+    const failedIndicator = await controller.findElement('[load-more-failed]');
+    expect(failedIndicator).to.not.be.null;
+    const failedIndicatorDisplay = await controller.getElementCssValue(
+        failedIndicator, 'display');
+    expect(failedIndicatorDisplay).to.equal('none');
+
+    const seeMoreButton = await controller.findElement('[load-more-button]');
+    expect(seeMoreButton).to.not.be.null;
+    const visibility = await controller.getElementCssValue(seeMoreButton,
+        'visibility', 'visible');
+    expect(visibility).to.equal('visible');
+    const display = await controller.getElementCssValue(seeMoreButton,
+        'display');
+    expect(display).to.equal('block');
```

Comment:
@cvializ Would it be useful to have a simple  `isDisplayed` method here? For example, the approach outlined here: https://w3c.github.io/webdriver/#dfn-bot-dom-isshown

Basically check if something has a rect (therefore has some sort of display, and all the parents do), check the opacity to see if it is actually visible (I think you need to check the ancestors too), check the computed visibility property and optionally check if it is hidden overflow from a parent element.

We could also start with something simple and add to it.

Reply:
+1! That would be nice indeed.

Reply:
I feel like that could be more than is needed, since our tests should be checking the state of the components they're testing. This feels like a heuristic more than a clear assertion and could introduce flakiness when people use it believing it to be more solid.

For example, is something visible if it meeds all the above criteria but is also `position: absolute; left: -9999`? What is the author of the code's intention? What if its opacity is .0001 for a technical reason like needing to be picked up by a screen reader?

I think for initial testing / proving it can be implemented as a helper method in test code, then be promoted to a `FunctionalTestController` API method after it is stable and well-defined. What do you think?

Reply:
Sounds good, we can start with just a helper function in this file that does what is needed. For the `position: absolute; left: -9999` case, screen readers will not be able to interact with that item, so I think that is also something we will want to check for. This is covered by the "hidden by overflow from a parent element", where the element is positioned in the overflow area of the body (otherwise the body would scroll horizontally).

Perhaps a better function would be `isInteractable` (which is less vague than `isDisplayed`), which would not check opacity, since that does not affect whether the element can receive interaction.